tesseract: 5.3.4 -> 5.4.1

21CSM opened the pull request:
## Description of changes
Bumps tesseract to 5.4.1.

I am also personally invested in this package, so adding myself as a maintainer. 

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

21CSM committed:
tesseract: 5.3.4 -> 5.4.1

Diff: https://github.com/tesseract-ocr/tesseract/compare/5.3.4...5.4.1

Changelog: https://github.com/tesseract-ocr/tesseract/blob/5.4.1/ChangeLog

kirillrdy commented on pkgs/applications/graphics/tesseract/tesseract5.nix:
```diff
@@ -5,13 +5,13 @@
     repo = "tesseract";
     rev = version;
-    sha256 = "sha256-IKxzDhSM+BPsKyQP3mADAkpRSGHs4OmdFIA+Txt084M=";
+    sha256 = "sha256-Yce9DVt1RJZkwN7ZlUE57eHm+cB9z7MbdFv8uCiGapo=";
```
```suggestion
    hash = "sha256-Yce9DVt1RJZkwN7ZlUE57eHm+cB9z7MbdFv8uCiGapo=";
```

21CSM replied:
Thanks. Addressed.

kirillrdy commented:
please consider adding link to Changelog and/or Diff to PR and commit message something like 3e051ceda2cbab6f4774387f32350eb0b47bd970

Scrumplex reviewed (commented):
It seems like OpenCL support was dropped in https://github.com/tesseract-ocr/tesseract/pull/4220. Removing `opencl-headers` from `buildInputs` still builds for me.

Edit: Building with and without `opencl-headers` seems to produce identical binaries, except for the different out path of course.

21CSM commented:
> It seems like OpenCL support was dropped in [tesseract-ocr/tesseract#4220](https://github.com/tesseract-ocr/tesseract/pull/4220). Removing `opencl-headers` from `buildInputs` still builds for me.
> 
> Edit: Building with and without `opencl-headers` seems to produce identical binaries, except for the different out path of course.

Thanks - it built for me and was able to test and works fine without opencl-headers. Addressed, and fixed some formatting.

Scrumplex commented:
Result of `nixpkgs-review pr 329842` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
  </ul>
</details>
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li>k2pdfopt</li>
    <li>libsForQt5.pix (plasma5Packages.pix)</li>
    <li>python311Packages.layoutparser</li>
    <li>python311Packages.layoutparser.dist</li>
    <li>python311Packages.pdf2docx</li>
    <li>python311Packages.pdf2docx.dist</li>
    <li>python312Packages.layoutparser</li>
    <li>python312Packages.layoutparser.dist</li>
    <li>python312Packages.pdf2docx</li>
    <li>python312Packages.pdf2docx.dist</li>
  </ul>
</details>
<details>
  <summary>73 packages built:</summary>
  <ul>
    <li>almanah</li>
    <li>arcan</li>
    <li>arcan-all-wrapped</li>
    <li>arcan-wrapped</li>
    <li>arcan.dev</li>
    <li>arcan.lib</li>
    <li>arcan.man</li>
    <li>cat9-wrapped</li>
    <li>durden-wrapped</li>
    <li>evolution</li>
    <li>evolution-ews</li>
    <li>evolutionWithPlugins</li>
    <li>gImageReader</li>
    <li>gnome-frog</li>
    <li>gscan2pdf</li>
    <li>gscan2pdf.man</li>
    <li>invoice2data</li>
    <li>invoice2data.dist</li>
    <li>kdePackages.skanpage</li>
    <li>kdePackages.skanpage.debug</li>
    <li>kdePackages.skanpage.dev</li>
    <li>kdePackages.skanpage.devtools</li>
    <li>libsForQt5.mauikit-imagetools (plasma5Packages.mauikit-imagetools)</li>
    <li>manga-cli</li>
    <li>mcomix</li>
    <li>mcomix.dist</li>
    <li>obs-studio-plugins.advanced-scene-switcher</li>
    <li>ocrmypdf (python312Packages.ocrmypdf)</li>
    <li>ocrmypdf.dist (python312Packages.ocrmypdf.dist)</li>
    <li>paperless-ngx</li>
    <li>pdfsandwich</li>
    <li>perl536Packages.ImageOCRTesseract</li>
    <li>perl536Packages.ImageOCRTesseract.devdoc</li>
    <li>perl538Packages.ImageOCRTesseract</li>
    <li>perl538Packages.ImageOCRTesseract.devdoc</li>
    <li>pipeworld-wrapped</li>
    <li>prio-wrapped</li>
    <li>python311Packages.llama-index</li>
    <li>python311Packages.llama-index-readers-file</li>
    <li>python311Packages.llama-index-readers-file.dist</li>
    <li>python311Packages.llama-index-readers-s3</li>
    <li>python311Packages.llama-index-readers-s3.dist</li>
    <li>python311Packages.llama-index.dist</li>
    <li>python311Packages.ocrmypdf</li>
    <li>python311Packages.ocrmypdf.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python311Packages.pymupdf</li>
    <li>python311Packages.pymupdf.dist</li>
    <li>python311Packages.pytesseract</li>
    <li>python311Packages.pytesseract.dist</li>
    <li>python311Packages.pytikz-allefeld</li>
    <li>python311Packages.pytikz-allefeld.dist</li>
    <li>python311Packages.videocr</li>
    <li>python311Packages.videocr.dist</li>
    <li>python312Packages.pymupdf</li>
    <li>python312Packages.pymupdf.dist</li>
    <li>python312Packages.pytesseract</li>
    <li>python312Packages.pytesseract.dist</li>
    <li>python312Packages.pytikz-allefeld</li>
    <li>python312Packages.pytikz-allefeld.dist</li>
    <li>python312Packages.videocr</li>
    <li>python312Packages.videocr.dist</li>
    <li>spamassassin</li>
    <li>spamassassin.devdoc</li>
    <li>termpdfpy</li>
    <li>termpdfpy.dist</li>
    <li>tesseract (tesseract5)</li>
    <li>textsnatcher</li>
    <li>tika</li>
    <li>vimPlugins.openscad-nvim</li>
    <li>xarcan</li>
    <li>zathura</li>
  </ul>
</details>

Scrumplex commented:
This actually seems to break `k2pdfopt` as it's trying to build a patched version of this derivation and fails to apply its patch.

Scrumplex commented:
Applying the following diff fixes k2pdfopt:

```diff
diff --git a/pkgs/applications/misc/k2pdfopt/default.nix b/pkgs/applications/misc/k2pdfopt/default.nix
index 32a0e31a315b..b0a8e5d17742 100644
--- a/pkgs/applications/misc/k2pdfopt/default.nix
+++ b/pkgs/applications/misc/k2pdfopt/default.nix
@@ -142,6 +142,9 @@ in stdenv.mkDerivation rec {
     };
     tesseract_modded = tesseract.override {
       tesseractBase = tesseract.tesseractBase.overrideAttrs ({ patches ? [], ... }: {
+        pname = "tesseract-k2pdfopt";
+        version = tesseract_patch.src.rev;
+        src = tesseract_patch.src;
         patches = patches ++ [ tesseract_patch ];
         # Additional compilation fixes
         postPatch = ''

```
CC @bosu @danielfullmer

21CSM commented:
Converted to draft @Scrumplex .
Will take a look at issues presented when I get some free time
